lib/ansible/module_utils/mysql.py

```diff
@@ -76,8 +79,30 @@ def mysql_connect(module, login_user=None, login_password=None, config_file='',
     if connect_timeout is not None:
         config['connect_timeout'] = connect_timeout
 
-    db_connection = mysql_driver.connect(**config)
+    try:
+        db_connection = mysql_driver.connect(**config)
+
+    except Exception as e:
+        module.fail_json(msg="unable to connect to database, check login_user and "
```

Comment:
Can't a network error occur there ? (wrong hostname / IP, network / DNS failure, etc ...)
If so, the error message should be more general and reflect the exception's message rather than adding extra verbosity which doesn't really explain why something went wrong

Reply:
Yes, you're absolutely right, thank you. I copy-pasted it from another mysql module and felt that something wrong when testing ;) Now I see, indeed, it should be more general. Done.